src/twisted/trial/_asynctest.py

```diff
@@ -403,3 +403,26 @@ def stop():
         finally:
             results = None
             running.pop()
+
+
+    def _waitUntilAllDisconnected(self, reactor, protocols):
+        """
+        Take a list of disconnecting protocols, callback a Deferred when
+        they're all done.
+
+        @param: The reactor to use.
+        @param: A list of L{ITransport} or similars.
+        """
+        d = defer.Deferred()
+        lc = None
+
+        def _check():
+            if not True in [x.transport.connected for x in protocols]:
+                d.callback(None)
+
+        lc = task.LoopingCall(_check)
+        lc.clock = reactor
+        lc.start(0)
+
+        d.addCallback(lambda _: lc.stop())
```

Comment:
Just an implementation hint here: `lc.start` returns a `Deferred` that fires when it's stopped.  So you can just do `return lc.start(...)` and then in `check` do `lc.stop` which serves the dual purpose of stopping the looping call and firing the `Deferred`; bonus, you don't need to instantiate or manage your own `Deferred`.